ramips: fix MTK_SOC for RT3662 devices

mans0n opened the pull request:
rt3883.mk contains both RT3662 and RT3883 device profiles, but commit
6a104ac77206 set MTK_SOC to rt3883 for all devices. This patch fixes it,
and renames dts files accordingly.

Fixes: 6a104ac77206 ("ramips/rt288x,rt3883: Name DTS files based on scheme")

mans0n commented:
I manually checked SoC numbers from:

asus,rt-n56u: internal photos
dlink,dir-645: internal photos
edimax,br-6475nd: OpenWrt Techdata, WikiDevi
loewe,wmdr-143n: initial commit
omnima,hpm: vendor webpage, initial commit
samsung,cy-swr1100: internal photos

belkin,f9k1109v1: initial commit
sitecom,wlr-6000: internal photos
trendnet,tew-691gr: internal photos
trendnet,tew-692gr: internal photos

adrianschmutzler commented:
Hi, when I did this in the first place, I decided upon names based on the compatible:
e.g. compatible = "asus,rt-n56u", "ralink,rt3883-soc";

So, should we change the compatible as well?

mans0n commented:
Oh, I didn't noticed that.
I made this commit because ath79 has both qca9557 and qca9558 ATH_SOC, and both dts files include qca9557.dtsi.
But now I found that not all compatible strings match their ATH_SOC. I'm split. 😕

mans0n commented:
It seems still most of the compatible strings do follow their ATH_SOC.
I think it's better to change the compatible to match MTK_SOC or ATH_SOC of the device, what do you think?

mans0n commented:
Added a commit to change compatible strings.

adrianschmutzler commented:
@chunkeey @ynezz Can one of you comment on whether we may change the compatible from ralink,rt3883-soc to ralink,rt3662-soc in this case.

hauke commented:
The compatible string in device tree should normally start with the most specific name and then you can add extra names like this:
```
compatible = "asus,rt-n56u", "ralink,rt3662-soc",  "ralink,rt3883-soc";
```
This says that this is a "asus,rt-n56u" is compatible to a "ralink,rt3662-soc", which is compatible to a  "ralink,rt3883-soc" (I do not know if this concrete example makes sense).

The kernel first checks if it can handle the first string, then the second and so on, this way we can use the last one in the beginning and when we later in the devices lifetime notice that this device needs some special handling we can do this based on the first entry. This is based on the idea that the device tree is shipped independent of the kernel and is stable which we do not really use in OpenWrt. 

See here for details:
https://elinux.org/Device_Tree_Usage#Understanding_the_compatible_Property

mans0n commented:
You're right, and adding a new soc compatible would not hurt.
But there's a list of possible ramips soc compatible strings in the kernel documentation,
and "ralink,rt3662-soc" is not in the list.
https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/mips/ralink.txt
Still we can use it independent of the kernel, though.
(like we use "mediatek,mt7628an-soc" instead of "ralink,mt7628a-soc")
What do you think about this?

mans0n commented:
Added ralink,rt3662-soc compatible strings instead of replacing ralink,rt3883-soc

adrianschmutzler commented:
Since both commits are symptoms of the same change, I would merge them into one commit (and maybe add a chatty comment on the compatible change to the commit message.

mans0n commented:
Okay, squashed.
Sorry, I can't think of a nice comment on this...

mans0n committed:
ramips: fix MTK_SOC for RT3662 devices

rt3883.mk contains both RT3662 and RT3883 device profiles, but commit
6a104ac77206 set MTK_SOC to rt3883 for all devices. This patch fixes it,
and renames dts files accordingly. And SoC compatible strings are also
appended in the dts.

Fixes: 6a104ac77206 ("ramips/rt288x,rt3883: Name DTS files based on scheme")

Signed-off-by: Sungbo Eo <mans0n@gorani.run>

adrianschmutzler commented:
Merged, thanks.
